block-layer/codecs/dag-jose.md

```diff
@@ -0,0 +1,126 @@
+# Specification: DAG-JOSE
+
+**Status: Descriptive - Draft**
+
+JOSE is a stanard for signing and encrypting JSON objects. The various specifications for JOSE can be found in the [IETF datatracker](https://datatracker.ietf.org/wg/jose/documents/). 
+
+DAG JOSE supports the full [IPLD Data Model](../data-model-layer/data-model.md) (within the payload).
+
+## Format
+
+The are two main ways to represent a JOSE object. As a JWS ([json web signature](https://datatracker.ietf.org/doc/rfc7515/?include_text=1)) and JWE ([json web encryption](https://datatracker.ietf.org/doc/rfc7516/?include_text=1)). These two formats  acts as the primitives in JOSE and can be used to create JWT and JWM objects etc. This specification describes how to encode JWS and JWE as an IPLD format.
+
+### Representation
+
+The layout of a decoded JOSE object is described by the IPLD schema defined below. We will refer to this layout as the `Decoded Representation`. 
+
+```ipldsch
+type Signature struct {
+  header optional {String:Any}
+  protected optional {String:Any}
+  signature Bytes
+}
+
+type JWS struct {
+  payload Any
```

Comment:
Maybe this should be a kinded union of `string` and `map`. The docs below suggest that if this this a string then it's base64url encoded bytes and if it's a map then it should be considered DAG-JSON. This seems to be an important distinction, no? Can it be anything other than these two, an `int`, `bool` or something else?

Reply:
Correct me if I’m wrong, but isn’t the payload the actual data that gets signed?

How is Any used here? Is the value *always* an encoded inline json/cbor structure?